src/App.jsx

```diff
@@ -1,8 +1,82 @@
+import {useState, useEffect} from "react";
+import {BrowserRouter, Routes, Route, Navigate} from "react-router-dom";
+import PublicRoute from "./pages/PublicRoute.jsx";
+import Login from "./pages/Login.jsx";
+import PrivateRoute from "./pages/PrivateRoute.jsx";
+import Dashboard from "./pages/Dashboard.jsx";
+import Register from "./pages/Register.jsx";
+import {fetchUser} from "./services/fetchUser.js";
+import {useDispatch, useSelector} from "react-redux";
+import {AuthProvider} from "./services/authContext.jsx";
 
+const App = () => {
+    const dispatch = useDispatch();
+    const user = useSelector((state) => state.auth.user);
+    const [isAuthenticated, setIsAuthenticated] = useState(false);
```

Comment:
in redux este state de isLogged in , nu e nevoie de use state aici, la fel si loading